DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc

```diff
@@ -0,0 +1,882 @@
+  LeadingTrailingCorrelationPerChannel = ibooker.book2D( "leading trailing correlation", title+" leading trailing correlation;leading edge (ns);trailing edge (ns)", 201, -100, 100, 201, -100, 100 );
+
+  ECCheckPerChannel = ibooker.book1D("optorxEC(8bit) - vfatEC vs optorxEC", title+" EC Error;optorxEC-vfatEC", 512, -256, 256 );
+
+  stripTomography_far = ibooker.book2D( "tomography far", "tomography with strips far;x + 50 OOT (mm);y (mm)", 50, 0, 50, 150, -50, 100 );
+  stripTomography_near = ibooker.book2D( "tomography near", "tomography with strips near;x + 50 OOT (mm);y (mm)", 50, 0, 50, 150, -50, 100 );
+  
+  hit_rate = ibooker.book1D( "hit rate", title+"hit rate;rate (Hz)", 1000, 0, 100 );
+}
+
+//----------------------------------------------------------------------------------------------------
+
+CTPPSDiamondDQMSource::CTPPSDiamondDQMSource( const edm::ParameterSet& ps ) :
+  tokenStatus_      ( consumes< edm::DetSetVector<TotemVFATStatus> >       ( ps.getParameter<edm::InputTag>( "tagStatus" ) ) ),
+  tokenLocalTrack_  ( consumes< edm::DetSetVector<TotemRPLocalTrack> >     ( ps.getParameter<edm::InputTag>( "tagLocalTrack" ) ) ),
+  tokenDigi_        ( consumes< edm::DetSetVector<CTPPSDiamondDigi> >      ( ps.getParameter<edm::InputTag>( "tagDigi" ) ) ),
+  tokenDiamondHit_  ( consumes< edm::DetSetVector<CTPPSDiamondRecHit> >    ( ps.getParameter<edm::InputTag>( "tagDiamondRecHits" ) ) ),
+  tokenDiamondTrack_( consumes< edm::DetSetVector<CTPPSDiamondLocalTrack> >( ps.getParameter<edm::InputTag>( "tagDiamondLocalTracks" ) ) ),
+  tokenFEDInfo_     ( consumes< std::vector<TotemFEDInfo> >                ( ps.getParameter<edm::InputTag>( "tagFEDInfo" ) ) ),
+  excludeMultipleHits_           ( ps.getParameter<bool>( "excludeMultipleHits" ) ),
+  minimumStripAngleForTomography_( ps.getParameter<double>( "minimumStripAngleForTomography" ) ),
+  verbosity_                     ( ps.getUntrackedParameter<unsigned int>( "verbosity", 0 ) ),
+  EC_difference_582_( -500 ), EC_difference_583_( -500 )
+{}
+
+//----------------------------------------------------------------------------------------------------
+
+CTPPSDiamondDQMSource::~CTPPSDiamondDQMSource()
+{}
+
+//----------------------------------------------------------------------------------------------------
+
+void
+CTPPSDiamondDQMSource::dqmBeginRun( const edm::Run&, const edm::EventSetup& )
+{}
+
+//----------------------------------------------------------------------------------------------------
+
+void
+CTPPSDiamondDQMSource::bookHistograms( DQMStore::IBooker& ibooker, const edm::Run&, const edm::EventSetup& )
+{
+  ibooker.cd();
+  ibooker.setCurrentFolder( "CTPPS" );
+  
+  globalPlot_= GlobalPlots( ibooker );
+
+  // loop over arms
+  for ( unsigned short arm = 0; arm < 2; ++arm ) {
+    // loop over stations
+    for ( unsigned short st = 1; st < 2; st += 2 ) {
+      // loop over RPs
+      for ( unsigned short rp = 6; rp < 7; ++rp ) {
```

Comment:
but its not the "same" code... you've copied it. Indeed, if this is a copy of some other code, can it just be merged and generalized via some input arguments?